Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -0,0 +1,455 @@
+        size_t size = 0;
+        for (const PathEntry& entry : m_entries)
+        {
+            ++size;
+            if (entry.IsEndOfArray())
+            {
+                size += 1;
+            }
+            else if (entry.IsIndex())
+            {
+                const size_t index = entry.GetIndex();
+                const double digitCount = index > 0 ? log10(aznumeric_cast<double>(index + 1)) : 1.0;
+                size += aznumeric_cast<size_t>(ceil(digitCount));
+            }
+            else
+            {
+                const char* nameBuffer = entry.GetKey().GetCStr();
+                for (size_t i = 0; nameBuffer[i]; ++i)
+                {
+                    if (nameBuffer[i] == EscapeCharacter || nameBuffer[i] == PathSeparator)
+                    {
+                        ++size;
+                    }
+                    ++size;
+                }
+            }
+        }
+        return size;
+    }
+
+    void Path::FormatString(char* stringBuffer, size_t bufferSize) const
+    {
+        size_t bufferIndex = 0;
+
+        auto putChar = [&](char c)
+        {
+            if (bufferIndex == bufferSize)
+            {
+                return;
+            }
+            stringBuffer[bufferIndex++] = c;
+        };
+
+        auto writeToBuffer = [&](const char* key)
+        {
+            for (size_t keyIndex = 0; key[keyIndex]; ++keyIndex)
+            {
+                const char c = key[keyIndex];
+                if (c == EscapeCharacter)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(TildeSequence);
+                }
+                else if (c == PathSeparator)
+                {
+                    putChar(EscapeCharacter);
+                    putChar(ForwardSlashSequence);
+                }
+                else
+                {
+                    putChar(c);
+                }
+            }
+        };
+
+        for (const PathEntry& entry : m_entries)
+        {
+            putChar(PathSeparator);
+            if (entry.IsEndOfArray())
+            {
+                putChar(EndOfArrayCharacter);
+            }
+            else if (entry.IsIndex())
+            {
+                bufferIndex += azsnprintf(&stringBuffer[bufferIndex], bufferSize - bufferIndex, "%zu", entry.GetIndex());
+            }
+            else
+            {
+                writeToBuffer(entry.GetKey().GetCStr());
+            }
+        }
+
+        putChar('\0');
+    }
+
+    AZStd::string Path::ToString() const
+    {
+        AZStd::string formattedString;
+        const size_t size = GetStringLength();
+        formattedString.resize_no_construct(size);
+        FormatString(formattedString.data(), size + 1);
+        return formattedString;
+    }
+
+    void Path::FromString(AZStd::string_view pathString)
+    {
+        m_entries.clear();
+        size_t pathIndex = 0;
+        bool isNumber = true;
+        for (size_t i = 0; i <= pathString.size(); ++i)
```

Comment:
You could just start the for loop at 1 and then set isNumber to the value of the previousIndex

This will get rid of the isNumber update at the end of the for loop.
Plus it will skip the useless first iteration.

i.e
```
for (size_t i = 1, pathIndex = 0; i <= pathString.size(); ++i)
{
    const bool isNumber = pathString[i -1] >= '0' && pathString[i - 1] <= '9';
   // ,,,
}
```

Reply:
`isNumber` is for the entirety of the entry (and shouldn't be cleared when a path separator is encountered), but I streamlined to `isNumber = isNumber && c >= '0' && c <= '9';`